src/include/storage/checkpoint_io.h

```diff
@@ -0,0 +1,254 @@
+#pragma once
+#include <vector>
+#include "storage/projected_columns.h"
+#include "storage/projected_row.h"
+#include "storage/write_ahead_log/log_io.h"
+
+namespace terrier::storage {
+// TODO(Zhaozhe): Should fetch block_size. Use magic number first.
+// The block size to output to disk every time
+#define CHECKPOINT_BLOCK_SIZE 4096
+
+/**
+ * The header of a page in the checkpoint file.
+ */
+// TODO(Zhaozhe, Mengyang): More fields can be added to header
+class CheckpointFilePage {
```

Comment:
If you're going to count on the data getting laid out as expected you may need to use the `PACKED` directive like is used for `ProjectedRow` (see include/storage/projected_row.h).

Reply:
Yes. We removed PACKED temporarily because of compilation error due to catalog::table_oid_t is a non-POD type. We might have to cast catalog::table_oid_t to POD type first to allow using PACKED.